### PR TITLE
LPS-154127 master

### PR DESCRIPTION
**Pull request by rolandpakai**
Hi @liferay-core-infra

could you please check my fix?
The _MARKER_INPUT_CLOSE is the >
See: https://www.w3schools.com/tags/tag_input.asp

Using the wrong value /> the processInput calculates wrong length value and not stripes the content.
https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/servlet/filters/strip/StripFilter.java#L396

Steps to Reproduce:

Add the below properties in portal-ext.properties.
com.liferay.portal.servlet.filters.strip.StripFilter=true
strip.mime.types=text/html*,text/xml*
Go to Design > Page Template and Create a new Page Template in the Masters section say "Master template".
3 Add a Documents and Media widget on either above or below the DropZone but not inside the dropzone. Click Publish.
Now go to Site Builder > Pages and Create a new Public Page based on the created "Master template"
Publish the page
Go to the created page
Right click on the page to "View Page Source".
Expected Result: The view page source should not have any gaps in between.
Actual Result: The view page source had many gaps in between.

Thanks, Roland

https://issues.liferay.com/browse/LPS-154127

cc @tinatian

### CODE REVIEW

**Comment by rolandpakai** [comment]
ci:test:stable

**Comment by rolandpakai** [comment]
ci:test:relevant

**Comment by rolandpakai** [comment]
ci:test:relevant

File: portal-impl/test/unit/com/liferay/portal/servlet/filters/strip/StripFilterTest.java [hunk 1]
```diff
@@ -494,6 +453,42 @@ public void testStrip() throws Exception {
...
 
 		Assert.assertEquals(21, charBuffer.position());
+
+		// Matching with br self-closing tag
+
+		charBuffer = CharBuffer.wrap("<input ><br />");
+
+		stringWriter = new StringWriter();
+
+		stripFilter.processInput(charBuffer, stringWriter);
```
Comment (dantewang) [inline portal-impl/test/unit/com/liferay/portal/servlet/filters/strip/StripFilterTest.java:463]: @rolandpakai If you are testing `strip` you should always call `strip`.

**Comment by dantewang** [comment]
Hi @rolandpakai, after looking into the `StripFilter` I think it makes sense to test `processInput` separately. I have left a message to Tina about my findings. Will let you know once we have a conclusion.

**Comment by rolandpakai** [comment]
Hi @dantewang

thanks for the review.

In my previouse PR i tested separatly the processInput but Tina asked me to not test processInput directly and move the test logic of testProcessInput to testStrip
https://github.com/liferay-core-infra/liferay-portal/pull/889#issuecomment-1150105601 

This is my 4th PR to liferay-core-infra and it seems this wont be merged neither becuase of contradictory request..

I would really appreciate if this PR could be forwarded without resending it.

Thanks, Roland

**Comment by dantewang** [comment]
@rolandpakai If we decide to go with the current test (testStrip), you need to modify your test to really call the strip method, instead of just moving previous logic in testProcessInput to testStrip.

If we decide that testProcessInput is OK, then we can just forward your first 2 commits.

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by dantewang** [comment]
@rolandpakai Hello Roland, I'll take over the test part and resend later. Does this one have any customer waiting on it?

**Comment by dantewang** [comment]
ci:close

**Pull request closed without merging**